src/cogent3/core/new_alignment.py

```diff
@@ -4468,6 +4474,29 @@ def count_gaps_per_seq(
 
         return result
 
+    def variable_positions(self, include_gap_motif: bool = True) -> list[int]:
+        """Return a list of variable position indexes.
+
+        Parameters
+        ----------
+        include_gap_motif
+            if False, sequences with a gap motif in a
+            column are ignored.
+
+        """
+        indices = []
```

Comment:
implement the following as a separate function outside the class
```python

@numba.jit
def _var_pos(arr, below):   # pragma: no cover
    m, n = arr.shape
    result = np.zeros(n, dtype=np.bool_)
    
    for col in range(n):
        last = -1
        for row in range(m):
            value = arr[row, col]
            if value < below:
                if last == -1:
                    last = value
                elif value != last:
                    result[col] = True
                    break
    
    return result
```
that will return a bool list much faster. Then return the indices for where True occurs.